src/m365/spfx/commands/spfx-doctor.ts

```diff
@@ -585,21 +595,41 @@ class SpfxDoctorCommand extends BaseProjectCommand {
 
       if (!spfxVersion) {
         logger.log(this.getStatus(CheckStatus.Failure, `SharePoint Framework`));
-        throw `SharePoint Framework not found`;
+        const message = 'SharePoint Framework not found';
+        this.resultsObject.push({
```

Comment:
You'll be logging an error here AND (in the finally-statement) output the resultsObject. I think we'd better take the route that was taken before on this topic: the SPFx check and version check aren't part of the fixes idea, they're evaluated separately and throw an error upfront. 

So you only need to add a line to the resultsObject if the checks pass. That way you at least have the information so you can log it at the end.